test/common.py

```diff
@@ -359,4 +360,115 @@ def setdata(self,data):
+                        0,  0,  0,  0,  0,   0,  90,  0,
+                        0,  0,  0, 10,  0,   0,   0,  9,
+                        1,  1,  1,  1,  2,   3,   4,  5,
+                        1,  2,  3,  4,  1,   2,   0,  0,
+                        0,  0,  0,  0,  0,   0,   0,  0,
+                        0,  0,  0,  0,  0,   0,   0,  0
+                       ]
+
+        self.green_pixels_1 = [
+                          11, 12, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0, 10, 2, 3, 4, 0,
+                           0,  0, 0,  0, 1, 0, 0, 0,
+                           0,  0, 1,  1, 2, 3, 4, 5,
+                           1,  2, 3,  4, 1, 2, 0, 0,
+                           0,  0, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0,  0, 1, 0, 0, 0
+                         ]
+
+        self.green_pixels_2 = [
+                          13, 12, 20,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  1,
+                           1,  0,  0,  0,  1,  32,   4,  2
+                          ]
+
+        self.blue_pixels_1 = [
+                         11, 12, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  1, 2, 3, 4, 5,
+                          1,  2, 3,  4, 1, 2, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 1
+                        ]
+
+        self.blue_pixels_2 = [
+                         16, 12, 20,  0,  0,   2,   3,  4,
+                          0,  0,  2,  3,  4,   5,   1,  0,
+                          0,  0,  0,  0,  0,   0,  90,  0,
+                          0,  0,  0, 10,  0,   0,   0,  9,
+                          1,  1,  1,  1,  2,   3,   4,  5,
+                          1,  2,  3,  4,  1,   2,   0,  1,
+                          1,  0,  0,  0,  0,   0,   0,  1,
+                          1,  0,  0,  0,  1,  32,   4,  2
+                        ]
+        # Get the output from the reference module.
+        self.output_red_pixels_1 = rle_code(self.red_pixels_1);
+
+    def setdata(self,data):
+        self.data = data
+        for i in range(64):
+            temp=self.data[i]
+            # Since the data we get contains amplitude, runlength as a single
+            # number therefore in order to extracts the two we take the modulus
+            # by 4096 to get last 12 bits and than shift and than again extracts
+            # the next 4 representing runlength.
```

Comment:
This doesn't explain why we want to modulus by 4096? Is this the size of the data? Is it got something to do with runlength?